src/core/server/http/router/router.ts

```diff
@@ -344,12 +350,12 @@ type RequestHandlerEnhanced<
  * @public
  */
 export type RequestHandler<
-  P extends ObjectType,
-  Q extends ObjectType,
-  B extends ObjectType | Type<Buffer> | Type<Stream>,
+  P extends ObjectType | ValidateFunction<unknown>,
+  Q extends ObjectType | ValidateFunction<unknown>,
```

Comment:
shouldn't we fallback to `= ValidateFunction<unknown>` to make all types optional? it would simplify manual declaration: 
```ts
const handler: RequestHandler<typeof validate.params> = (context, req, res) => { ... };
```